### PR TITLE
Implemented Irv Application

### LINKED ISSUE #36: Import style layer from qgis plugin
For the import project from IRMT (plugin qgis) to platform there is a problem with import the style.
To reproduce this problem:

Install Platform from branch pla26-irv and start
Open qgis
Connect to platform (IRMT settings (settings icon) -> create connection to platform)
Menu IRMT -> oq platform -> Load socioeconomic data -> get indicator -> select indicators and countries
Save project
Click weight data (balance icon) and save
Menu IRMT -> oq platform -> upload project to platform

### PR DESCRIPTION
**Pull request by hascar**
Implemented Irv Application. 
In the same time, did some changes in IRMT plugin of qgis
The tests are green here: https://ci.openquake.org/job/zdevel_oq-platform2/517/
Close #36

### CODE REVIEW

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 1]
```diff
@@ -0,0 +1,2038 @@
...
+        IndicatorObj[tempKeys[i]] = scaledValues[i];
+    }
+    return IndicatorObj;
+}
+
+// Mapbox-gl stuff
+function setupMapboxGlMap() {
+
+    // Create mapbox map element
+    mapboxgl.accessToken = 'pk.eyJ1IjoiZ2VtZGV2IiwiYSI6ImNqbG1kamlmYTE3MHUzdWxkZWEweHRoeHUifQ.wErApo8meMm031A9a2n26Q';
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Changed acces_token mapbox

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 2]
```diff
@@ -0,0 +1,2038 @@
...
+        }
+    }
+    return false;
+}
+
+function getGeoServerLayers() {
+    $('#load-project-spinner').show();
+    var IRMTLayerNames = [];
+    var IRMTLayerTitle = [];
+    var url = "http://"+ window.location.hostname +":8000/geoserver/ows?service=WFS&version=1.0.0&REQUEST=GetCapabilities&SRSNAME=EPSG:4326&outputFormat=json&format_options=callback:getJson";
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added IRMTLayerTitle
Changed path ows, before was : 
var url = "/geoserver/ows?service=WFS&version=1.0.0&REQUEST=GetCapabilities&SRSNAME=EPSG:4326&outputFormat=json&format_options=callback:getJson";

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 3]
```diff
@@ -0,0 +1,2038 @@
...
+            //convert XML to JSON
+            var xmlText = new XMLSerializer().serializeToString(xml);
+            var x2js = new X2JS();
+
+            var jsonElement = x2js.xml_str2json(xmlText);
+
+            var featureType = jsonElement.WFS_Capabilities.FeatureTypeList.FeatureType;
+
+            // Find the IRMT keywords
+            var stringsToLookFor = ['SVIR', 'IRMT_QGIS_Plugin'];
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Changed this.
Before was: 
var stringsToLookFor = ['SVIR_QGIS_Plugin', 'IRMT_QGIS_Plugin'];

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 4]
```diff
@@ -0,0 +1,2038 @@
...
+                return;
+            }
+
+            for (var i = 0; i < featureType.length; i++) {
+                for (var j=0; j < stringsToLookFor.length; j++) {
+                    stringToLookFor = stringsToLookFor[j];
+                    if (featureType[i].Keywords.indexOf(stringToLookFor) > -1) {
+                        // IRMTLayerNames.push(featureType[i].Title + " (" + featureType[i].Name + ")");
+                        IRMTLayerNames.push(featureType[i].Name);
+                        IRMTLayerTitle.push(featureType[i].Title);
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added this.

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 5]
```diff
@@ -0,0 +1,2038 @@
...
+		            '<p>No layer found.</p>'
+		        );
+		        $('#ajaxErrorDialog').dialog('open');
+            }
+
+            // Create AngularJS dropdown menu
+            var mapLayerList = [];
+            for (var ij = 0; ij < IRMTLayerNames.length; ij++) {
+                var Obj = {};
+                Obj.name = IRMTLayerNames[ij];
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Changed this.
Before was :  tempObj.name = IRMTLayerNames[ij];

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 6]
```diff
@@ -0,0 +1,2038 @@
...
+                Obj.title = IRMTLayerTitle[ij];
+
+                $('#layer-list').append('<div id="list' + ij +'">' + Obj.title + '</div>');
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added this.

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 7]
```diff
@@ -0,0 +1,2038 @@
...
+            // Create AngularJS dropdown menu
+            var mapLayerList = [];
+            for (var ij = 0; ij < IRMTLayerNames.length; ij++) {
+                var Obj = {};
+                Obj.name = IRMTLayerNames[ij];
+                Obj.title = IRMTLayerTitle[ij];
+
+                $('#layer-list').append('<div id="list' + ij +'">' + Obj.title + '</div>'); 
+
+                $('#list'+ ij).on('click', function () {
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added this.

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 8]
```diff
@@ -0,0 +1,2038 @@
...
+                    window.location = 'http://' + location.hostname + ':8000/irv/' + Obj.name;
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added this .

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 9]
```diff
@@ -0,0 +1,2038 @@
...
+
+        widget.draggable({
+            stack: "div",  // put on top of the others when dragging
+            distance: 0,   // do it even if it's not actually moved
+            cancel: "#project-def, #primary-chart, #cat-chart, #iri-chart, #iri-chart, #themeSelector"
+        });
+        widget.css({
+            'display': 'none',
+            // perhaps we should not set width and height here, and let things automatically resize
+            'width': '860px',
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Changed this.
Before was: 'width': '700px',

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 10]
```diff
@@ -0,0 +1,2038 @@
...
+    $('#map-tools').append(
+        '<div class="l_proj"><button id="loadProjectdialogBtn" type="button" class="btn btn-blue">Load Project</button></div>'
+    );
+    // Using the list in order to keep the order
+    $.each(['indicators', 'svi', 'iri', 'projDef'], function(i, widgetName) {
+        // slice(1) removes the heading # from the selector
+        var buttonId = widgetsAndButtons[widgetName].button.slice(1);
+        var buttonText = widgetsAndButtons[widgetName].buttonText;
+        $('#map-tools').append(
+            '<div class="but_l_proj"><button id="' + buttonId + '" type="button" class="btn btn-blue" disabled>' + buttonText + '</button></div>'
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added div with class "but_l_proj"

File: openquakeplatform/irv/static/irv/js/irv_viewer.js [hunk 11]
```diff
@@ -0,0 +1,2038 @@
...
+            $(widgetAndBtn.button).css({
+                'background': '#f7931e'
+            });
+        } else {                              // Change Show -> Hide
+            btnText = 'Hide ' + btnText.slice(5);
+            $(widgetAndBtn.button).css({
+                'background': '#f05a24'
+            });
+        }
+
```
Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added class "but_l_proj" style

Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Added class "l_proj" and "but_l_proj" styles

Comment (hascar) [inline openquakeplatform/irv/static/irv/js/irv_viewer.js]: Changed this path.
Before was: url: '/geoserver/oqplatform/ows?service=WFS&version=1.0.0&request=GetFeature&typeName='+ selectedLayer +'&outputFormat=json',

File: openquakeplatform/irv/static/irv/js/irv_viewer.js (hunk 3 above)
Comment (ptormene) [reply openquakeplatform/irv/static/irv/js/irv_viewer.js]: I suppose it was safer as it was before. Why did you change this?

File: openquakeplatform/irv/static/irv/js/irv_viewer.js (hunk 5 above)
Comment (ptormene) [reply openquakeplatform/irv/static/irv/js/irv_viewer.js]: I don't like the capital letter in Obj (and actually also the name is too generic).
I suggest to rename all occurrences of `Obj` into something like `layerProperties`

**Comment by ptormene** [comment]
Besides minor comments, I would like to see the application working, before giving the LGTM

**Pull request merged**